libkbfs/folder_branch_ops.go

```diff
@@ -2522,13 +2483,22 @@ func (fbo *folderBranchOps) createEntryLocked(
 		Context: kbfsblock.MakeFirstContext(
 			session.UID, keybase1.BlockType_DATA),
 	}
+	co.AddRefBlock(newPtr)
+	co.AddUpdate(parentPtr, parentPtr)
```

Comment:
The "updates" between the same pointers are just placeholders that will be fixed up by the folder update prep process in `syncAllLocked`.  It tells that process what blocks to ready, and the final pointer gets replaced with the new ID at the end of the process.